File: 35-search-insert-position/35-search-insert-position.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        hold=[]
        
        if(target in nums):
            return (nums.index(target))
        
        if(target>nums[-1]):
            hold3=nums.index(nums[-1])
            hold3=hold3+1
            return hold3
    
            
        hold=nums
        hold.append(target)
        hold.sort()
        
        return (hold.index(target))
                
                
                            
                            
        

File: 35-search-insert-position/35-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        
        
        
        int f=0;
        int l=nums.size()-1;
        int m;
        while(f<=l)
        {
            
            m=(f+l)/2;
            if(target==nums[m])
            {
                return m;
            }
            
            else if(target<nums[m])
            {
                l=m-1; 
                
                
            }
            else if(target>nums[m])
            {
                f=m+1;
                
            }            
        }
        return l+1;
        
    }
};